### PR TITLE
backend: CLI config updates

### PR DESCRIPTION
**Pull request by tianjing-li**
Thank you for contributing to the Cohere Toolkit!

- [ ] **PR title**: "area: description"
  - Where "area" is whichever of interface, frontend, model, tools, backend, etc. is being modified. Use "docs: ..." for purely docs changes, "infra: ..." for CI changes.
  - Example: "deployment: add Azure model option"

- [ ] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change

- [ ] **Add tests and docs**: Please include testing and documentation for your changes
- [ ] **Lint and test**: Run `make lint` and `make run-tests`

**AI Description**

The pull request introduces changes to the Cohere Toolkit's backend, specifically enhancing the setup process and configuration.

**Files Affected:**
- `src/backend/cli/constants.py`
- `src/backend/cli/main.py`
- `src/backend/cli/prompts.py` (new file)
- `src/backend/cli/setters.py` (new file)
- `src/backend/cli/utils.py` (new file)
- `src/backend/config/settings.py`

**Changes Made:**
- Refactored code to improve organization and modularity:
   - Introduced new modules: `prompts.py`, `setters.py`, and `utils.py`.
   - Moved relevant functions to the new modules.
- Enhanced the setup process:
   - Added prompts for database URL, Redis URL, public backend API hostname, and frontend client hostname.
   - Introduced a deployment prompt to allow users to select model deployments.
   - Updated the welcome message and wrap-up instructions.
- Improved configuration handling:
   - Introduced YAML configuration files for better organization of settings.
   - Updated the `SETTINGS_CONFIG` class to manage configuration sources.
   - Modified the `Settings` class to use `SETTINGS_CONFIG`.
- Minor adjustments to function calls and imports.

### CODE REVIEW

File: src/backend/cli/constants.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
+from enum import StrEnum
```
Comment (tianjing-li) [inline src/backend/cli/constants.py:1]: Couple refactorings:

Split up the code into
- constants
- prompts
- utils
- setters

Comment (tianjing-li) [reply src/backend/cli/constants.py:1]: Most of the new code is in setters (setting YAML config)

File: src/backend/cli/setters.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import os
```
Comment (tianjing-li) [inline src/backend/cli/setters.py:1]: New code here

File: src/backend/config/settings.py [hunk 3]
```diff
@@ -10,21 +10,34 @@
...
-setting_config = SettingsConfigDict(
+SETTINGS_CONFIG = SettingsConfigDict(
     env_file=".env",
     extra="ignore",
     env_nested_delimiter="_",
     env_prefix="",
     env_ignore_empty=True,
 )
 
+CONFIG_PATH = "src/backend/config"
```
Comment (tianjing-li) [inline src/backend/config/settings.py:21]: Small refactor to just convert these constants into caps, in line with the rest of the codebase

File: src/backend/cli/constants.py (hunk 1 above)
Comment (lusmoura) [reply src/backend/cli/constants.py:1]: Nice!

File: src/backend/config/configuration.template.yaml [hunk 4]
```diff
@@ -5,8 +5,6 @@ deployments:
     - sagemaker
     - azure
     - bedrock
-  cohere_platform:
```
Comment (lusmoura) [inline src/backend/config/configuration.template.yaml:8]: Why are we removing this?

**Review by lusmoura — APPROVED** [review]

File: src/backend/config/configuration.template.yaml (hunk 4 above)
Comment (tianjing-li) [reply src/backend/config/configuration.template.yaml:8]: It's duplicated - there's the same setting in the secrets template and I figured it's more appropriate there

**Pull request merged**